### PR TITLE
Magento_ProductAlert module. Replace replace deprecated message functions

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

PR replaces deprecated message functions in the `Magento_ProductAlert` module.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/ProductAlert/Controller/Add/Price.php [hunk 1]
```diff
@@ -6,16 +6,18 @@
...
 namespace Magento\ProductAlert\Controller\Add;
 
-use Magento\Framework\App\Action\HttpGetActionInterface;
-use Magento\ProductAlert\Controller\Add as AddController;
-use Magento\Framework\App\Action\Context;
-use Magento\Customer\Model\Session as CustomerSession;
-use Magento\Store\Model\StoreManagerInterface;
 use Magento\Catalog\Api\ProductRepositoryInterface;
-use Magento\Framework\UrlInterface;
+use Magento\Customer\Model\Session as CustomerSession;
```
Comment (ihor-sviziev) [inline app/code/Magento/ProductAlert/Controller/Add/Price.php:10]: Could you replace `\Magento\Customer\Model\Session` to `CustomerSession` in all places, for instance in constructor's phpdoc block

The same with all other imported classes

File: app/code/Magento/ProductAlert/Controller/Add/Price.php [hunk 2]
```diff
@@ -6,16 +6,18 @@
...
 use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpGetActionInterface;
 use Magento\Framework\Controller\ResultFactory;
+use Magento\Framework\Controller\Result\Redirect;
 use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\UrlInterface;
+use Magento\ProductAlert\Controller\Add as AddController;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Tests\NamingConvention\true\string;
```
Comment (ihor-sviziev) [inline app/code/Magento/ProductAlert/Controller/Add/Price.php]: This is mistake, please remove it.

Note: it's better to ignore `vendor/magento/magento-coding-standard` folder in PhpStorm in order not to have such mistakes

File: app/code/Magento/ProductAlert/Controller/Add/Price.php [hunk 3]
```diff
@@ -68,13 +71,14 @@ protected function isInternal($url)
     /**
      * Method for adding info about product alert price.
      *
-     * @return \Magento\Framework\Controller\Result\Redirect
+     * @return \Magento\Framework\App\ResponseInterface|Redirect|\Magento\Framework\Controller\ResultInterface
```
Comment (ihor-sviziev) [inline app/code/Magento/ProductAlert/Controller/Add/Price.php]: I see that currently we could have only `\Magento\Framework\Controller\ResultInterface`. `Magento\Framework\App\ResponseInterface` is never returned here, `Redirect` implements `ResponseInterface`

File: app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php [hunk 4]
```diff
@@ -51,8 +56,13 @@ public function execute()
             /* @var $product \Magento\Catalog\Model\Product */
             $product = $this->productRepository->getById($productId);
             if (!$product->isVisibleInCatalog()) {
-                throw new NoSuchEntityException();
+                $this->messageManager->addErrorMessage(
```
Comment (ihor-sviziev) [inline app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php:60]: It looks like separate feature. Would be great to cover it with tests.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php (hunk 4 above)
Comment (atwixfirster) [reply app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php:60]: > It looks like separate feature. Would be great to cover it with tests.

We do not have unit tests for any of `Magento_ProductAlert` controllers.

Probably now you will ask me to create Unit Tests for each ones....

File: app/code/Magento/ProductAlert/Controller/Add/Price.php (hunk 2 above)
Comment (atwixfirster) [reply app/code/Magento/ProductAlert/Controller/Add/Price.php]: **done**

File: app/code/Magento/ProductAlert/Controller/Add/Price.php (hunk 1 above)
Comment (atwixfirster) [reply app/code/Magento/ProductAlert/Controller/Add/Price.php:10]: **done**

File: app/code/Magento/ProductAlert/Controller/Add/Price.php (hunk 3 above)
Comment (atwixfirster) [reply app/code/Magento/ProductAlert/Controller/Add/Price.php]: Good point. 

**implemented**

Thank you, @ihor-sviziev

File: app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php (hunk 4 above)
Comment (ihor-sviziev) [reply app/code/Magento/ProductAlert/Controller/Unsubscribe/Price.php:60]: Ok, let's leave it as is for now

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @atwixfirster,
Could you fix static tests?

**Comment by atwixfirster** [comment]
@ihor-sviziev 

> Hi @atwixfirster,
> Could you fix static tests?

**done**

All tests have been successfully passed.

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-5862](https://jira.corp.magento.com/browse/ENGCOM-5862) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**